### DISCUSSION TITLE
Which Legacy Arduino IDE will chipKIT-core work with?

### DISCUSSION THREAD

**Discussion by djsbriscoe** [chipKIT32/chipKIT-core#485] [General]
I've tried to use chipKIT-core with Arduino IDE 2.04, and I get the error: Failed to install platform: chipKIT:pic32.
Error: 13 INTERNAL: Cannot install tool chipKIT:pic32-tools@1.43-pic32gcc: searching package root dir: no unique root dir in archive, found 'C:\Users\david\AppData\Local\Arduino15\tmp\package-2749070939\bin' and 'C:\Users\david\AppData\Local\Arduino15\tmp\package-2749070939\lib'
Apparently the latest IDE is too strict and the chipKIT-core will NOT work with the latest IDE. Which LEGACY VERSION of the Arduino IDE will chipKIT-core work with? I have a chipKIT uC32 and a Uno32 board that I'd like to make use of. Thanks.

**Comment by majenkotech** [comment]
This is a known problem (and one we discussed in a meeting earlier today as it happens). It's simply that the compiler isn't packaged in quite the right way for the newer IDEs (and for `arduino-cli` too). 

Since I use `arduino-cli` all the time, in order to get it working I had to re-package the compilers (simply unzip into a folder then re-zip that folder), which of course meant a new .json file with the new SHA checksums and such in.

It should work out of the box with any 1.x IDE later than (IIRC) 1.6.2, or you can use my patched version to use 2.x or `arduino-cli`, the json for which is here:

* https://majenko.co.uk/i/package_chipkit_index.json

**Comment by JacobChrist** [comment]
I use it with 1.8.19 without issue.

**Comment by djsbriscoe** [comment]
Thanks for your replies.
I have tried the following
1/ I have temporarily disabled the sandbox function of my Antivirus software. The files produced where being quarantined.
2/ I have given write permissions to the folders mentioned in the directories mentioned in the error message below. I will try to give ownership as well.
The errors I'm getting are as follows for the BASIC "BLINK" example.
`Using board 'chipkit_uc32' from platform in folder: C:\Users\david\AppData\Local\Arduino15\packages\chipKIT\hardware\pic32\2.1.0
Using core 'pic32' from platform in folder: C:\Users\david\AppData\Local\Arduino15\packages\chipKIT\hardware\pic32\2.1.0
Detecting libraries used...
"C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\tools\\pic32-tools\\1.43-4-pic32gcc/bin/pic32-g++" -c -g -O2 -w -std=gnu++11 -DARDUINO_ARCH_PIC32 -mno-smart-io -ffunction-sections -fdata-sections -mdebugger -Wcast-align -fno-short-double -ftoplevel-reorder -fno-exceptions -w -x c++ -E -CC -mprocessor=32MX340F512H -DF_CPU=80000000L -DARDUINO=10607 -D_BOARD_UC32_ -DARDUINO_ARCH_PIC32 -DMPIDEVER=16777998 -DMPIDE=150 -DIDE=Arduino -G1024 "-IC:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\0A76FD69009E1A16DCEADBB98A6101E7/sketch" "-IC:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\cores\\pic32" "-IC:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\variants\\uC32" "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\0A76FD69009E1A16DCEADBB98A6101E7\\sketch\\Blink.ino.cpp" -o nul
Generating function prototypes...
"C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\tools\\pic32-tools\\1.43-4-pic32gcc/bin/pic32-g++" -c -g -O2 -w -std=gnu++11 -DARDUINO_ARCH_PIC32 -mno-smart-io -ffunction-sections -fdata-sections -mdebugger -Wcast-align -fno-short-double -ftoplevel-reorder -fno-exceptions -w -x c++ -E -CC -mprocessor=32MX340F512H -DF_CPU=80000000L -DARDUINO=10607 -D_BOARD_UC32_ -DARDUINO_ARCH_PIC32 -DMPIDEVER=16777998 -DMPIDE=150 -DIDE=Arduino -G1024 "-IC:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\0A76FD69009E1A16DCEADBB98A6101E7/sketch" "-IC:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\cores\\pic32" "-IC:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\variants\\uC32" "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\0A76FD69009E1A16DCEADBB98A6101E7\\sketch\\Blink.ino.cpp" -o "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\0A76FD69009E1A16DCEADBB98A6101E7\\preproc\\ctags_target_for_gcc_minus_e.cpp"
"C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\builtin\\tools\\ctags\\5.8-arduino11/ctags" -u --language-force=c++ -f - --c++-kinds=svpf --fields=KSTtzns --line-directives "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\0A76FD69009E1A16DCEADBB98A6101E7\\preproc\\ctags_target_for_gcc_minus_e.cpp"
Compiling sketch...
"C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\tools\\pic32-tools\\1.43-4-pic32gcc/bin/pic32-g++" -c -g -O2 -w -std=gnu++11 -DARDUINO_ARCH_PIC32 -mno-smart-io -ffunction-sections -fdata-sections -mdebugger -Wcast-align -fno-short-double -ftoplevel-reorder -fno-exceptions -mprocessor=32MX340F512H -DF_CPU=80000000L -DARDUINO=10607 -D_BOARD_UC32_ -DMPIDEVER=16777998 -DMPIDE=150 -DIDE=Arduino -G1024 "-IC:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\0A76FD69009E1A16DCEADBB98A6101E7/sketch" "-IC:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\cores\\pic32" "-IC:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\variants\\uC32" "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\0A76FD69009E1A16DCEADBB98A6101E7\\sketch\\Blink.ino.cpp" -o "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\0A76FD69009E1A16DCEADBB98A6101E7\\sketch\\Blink.ino.cpp.o"
Compiling libraries...

Compiling core...
"C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\tools\\pic32-tools\\1.43-4-pic32gcc/bin/pic32-gcc" -c -g -O2 -w -std=gnu11 -DARDUINO_ARCH_PIC32 -mno-smart-io -ffunction-sections -fdata-sections -mdebugger -Wcast-align -fno-short-double -ftoplevel-reorder -MMD -mprocessor=32MX340F512H -DF_CPU=80000000L -DARDUINO=10607 -D_BOARD_UC32_ -DMPIDEVER=16777998 -DMPIDE=150 -DIDE=Arduino -G1024 "-IC:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\0A76FD69009E1A16DCEADBB98A6101E7/sketch" "-IC:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\cores\\pic32" "-IC:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\variants\\uC32" "C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\variants\\uC32\\Board_Data.c" -o "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\0A76FD69009E1A16DCEADBB98A6101E7\\core\\Board_Data.c.o"
exit status 255

Compilation error: exit status 255`

There is nothing being written to the 
"C:\Users\david\AppData\Local\Temp\arduino\sketches\0A76FD69009E1A16DCEADBB98A6101E7\core" folder. I will try to give the write permissions to this folder and see if it makes a difference.
I've tried several times to edit the permissions of the folders mentioned and they keep going back to "READ ONLY" no matter what I do. I don't know what I'm doing wrong.

I will look up "Exit Status 255"

Where can I find information on setting up and using the arduino-cli? I'd like to learn how to use it for tracking down this problem. Thanks.

David.

**Comment by djsbriscoe** [comment]
How can the compiler be invoked directly from the command line for the chipKIT? Where is the executable for the compiler located so that I can put it into my system path (if it's not already there)? Thanks.

**Comment by majenkotech** [comment]
Did you download the IDE from the Arduino website, or install it from the Windows App Store? I have seen problems in the past with the Windows App Store version that look similar to this, although that was a long time ago and I have no idea if they still persist.

The compiler is as stated in your output above: `C:\Users\david\AppData\Local\Arduino15\packages\chipKIT\tools\pic32-tools\1.43-4-pic32gcc/bin/pic32-gcc` (and `...-g++` of course), although what the exit status of 255 signifies I have no idea.

**Reply by djsbriscoe** [reply]
I downloaded V2 of the Arduino IDE whilst it was in Release candidate stage a while ago I think and it has been updating ever since. I'll uninstal ALL of my Arduino IDE installations (and save the .json file links) and reinstall and try again with a fresh install of the IDE. I'll report back later. I REALLY want to get these boards working as I'm trying to increase my knowledge of the PIC32 chips and chipKIT is a convenient start.

**Reply by majenkotech** [reply]
Which version of Windows are you using? I haven't actually tested my repacked version of the compiler on windows yet since I don't use it, so I'll try setting up the ide on the same version of Windows that you are using and installing my repacked version to see what happens. It could be that something got broken during the repacking or some file got missed or corrupted.

**Reply by djsbriscoe** [reply]
Success.
'Using board 'chipkit_uc32' from platform in folder: C:\Users\david\AppData\Local\Arduino15\packages\chipKIT\hardware\pic32\2.1.0
Using core 'pic32' from platform in folder: C:\Users\david\AppData\Local\Arduino15\packages\chipKIT\hardware\pic32\2.1.0
Detecting libraries used...
"C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\tools\\pic32-tools\\1.43-4-pic32gcc/bin/pic32-g++" -c -g -O2 -w -std=gnu++11 -DARDUINO_ARCH_PIC32 -mno-smart-io -ffunction-sections -fdata-sections -mdebugger -Wcast-align -fno-short-double -ftoplevel-reorder -fno-exceptions -w -x c++ -E -CC -mprocessor=32MX340F512H -DF_CPU=80000000L -DARDUINO=10607 -D_BOARD_UC32_ -DARDUINO_ARCH_PIC32 -DMPIDEVER=16777998 -DMPIDE=150 -DIDE=Arduino -G1024 "-IC:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E/sketch" "-IC:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\cores\\pic32" "-IC:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\variants\\uC32" "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E\\sketch\\Blink.ino.cpp" -o nul
Generating function prototypes...
"C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\tools\\pic32-tools\\1.43-4-pic32gcc/bin/pic32-g++" -c -g -O2 -w -std=gnu++11 -DARDUINO_ARCH_PIC32 -mno-smart-io -ffunction-sections -fdata-sections -mdebugger -Wcast-align -fno-short-double -ftoplevel-reorder -fno-exceptions -w -x c++ -E -CC -mprocessor=32MX340F512H -DF_CPU=80000000L -DARDUINO=10607 -D_BOARD_UC32_ -DARDUINO_ARCH_PIC32 -DMPIDEVER=16777998 -DMPIDE=150 -DIDE=Arduino -G1024 "-IC:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E/sketch" "-IC:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\cores\\pic32" "-IC:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\variants\\uC32" "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E\\sketch\\Blink.ino.cpp" -o "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E\\preproc\\ctags_target_for_gcc_minus_e.cpp"
"C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\builtin\\tools\\ctags\\5.8-arduino11/ctags" -u --language-force=c++ -f - --c++-kinds=svpf --fields=KSTtzns --line-directives "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E\\preproc\\ctags_target_for_gcc_minus_e.cpp"
Compiling sketch...
"C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\tools\\pic32-tools\\1.43-4-pic32gcc/bin/pic32-g++" -c -g -O2 -w -std=gnu++11 -DARDUINO_ARCH_PIC32 -mno-smart-io -ffunction-sections -fdata-sections -mdebugger -Wcast-align -fno-short-double -ftoplevel-reorder -fno-exceptions -mprocessor=32MX340F512H -DF_CPU=80000000L -DARDUINO=10607 -D_BOARD_UC32_ -DMPIDEVER=16777998 -DMPIDE=150 -DIDE=Arduino -G1024 "-IC:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E/sketch" "-IC:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\cores\\pic32" "-IC:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\variants\\uC32" "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E\\sketch\\Blink.ino.cpp" -o "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E\\sketch\\Blink.ino.cpp.o"
Compiling libraries...
Compiling core...
Using previously compiled file: C:\Users\david\AppData\Local\Temp\arduino\sketches\8497FD2078FAD83AECF9D6491A36476E\core\Board_Data.c.o
Using precompiled core: C:\Users\david\AppData\Local\Temp\arduino\cores\chipKIT_pic32_chipkit_uc32_PMOD_no_d3e01e7679fb639b6ccf50735c4d7b54\core.a
Linking everything together...
"C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\tools\\pic32-tools\\1.43-4-pic32gcc/bin/pic32-g++" -w -Os "-Wl,--save-gld=C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E/sketch.ld,-Map=C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E/sketch.map,--gc-sections" -mdebugger -mno-peripheral-libs -nostartfiles -mprocessor=32MX340F512H -o "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E/Blink.ino.elf" "C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\cores\\pic32/cpp-startup.S" "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E\\sketch\\Blink.ino.cpp.o" "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E\\core\\Board_Data.c.o" "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E/..\\..\\cores\\chipKIT_pic32_chipkit_uc32_PMOD_no_d3e01e7679fb639b6ccf50735c4d7b54\\core.a" "-LC:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E" -lm -T "C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\cores\\pic32/chipKIT-application-32MX340F512.ld" -T "C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\hardware\\pic32\\2.1.0\\cores\\pic32/chipKIT-application-COMMON.ld"
"C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\tools\\pic32-tools\\1.43-4-pic32gcc/bin/pic32-objcopy" -O ihex -j .eeprom --set-section-flags=.eeprom=alloc,load --no-change-warnings --change-section-lma .eeprom=0 "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E/Blink.ino.elf" "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E/Blink.ino.eep"
"C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\tools\\pic32-tools\\1.43-4-pic32gcc/bin/pic32-bin2hex" -a "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E/Blink.ino.elf"

"C:\\Users\\david\\AppData\\Local\\Arduino15\\packages\\chipKIT\\tools\\pic32-tools\\1.43-4-pic32gcc/bin/pic32-size" -A "C:\\Users\\david\\AppData\\Local\\Temp\\arduino\\sketches\\8497FD2078FAD83AECF9D6491A36476E/Blink.ino.elf"
Sketch uses 5476 bytes (1%) of program storage space. Maximum is 520192 bytes.
Global variables use 3072 bytes (18%) of dynamic memory, leaving 13312 bytes for local variables. Maximum is 16384 bytes.
"C:\Users\david\AppData\Local\Arduino15\packages\chipKIT\tools\pic32prog\v2.1.46/pic32prog.exe"   -d "COM9"  -b 115200  "C:\Users\david\AppData\Local\Temp\arduino\sketches\8497FD2078FAD83AECF9D6491A36476E/Blink.ino.hex"
    Copyright: (C) 2011-2015 Serge Vakulenko
                   2016-2017 Majenko Technologies
Programmer for Microchip PIC32 microcontrollers, Version 2.1.46
      Adapter: STK500v2 Bootloader
 Program area: 1d000000-1d1fffff
    Processor: Bootloader
 Flash memory: 2048 kbytes
  Boot memory: 80 kbytes
         Data: 6780 bytes
        Erase: done
Program flash: .........########## done
 Verify flash: .........######### done
 Program rate: 2285 bytes per second
`
I'm using Windows 10 V22H2 (OS Build 19045.2728).

I did a clear out of the User data temp folder as well as reinstalling Arduino V2.04 and the picKIT library from the link you gave me. I'll try it on the Uno32 board as well. If there are any problems I'll report back. Thanks.

**Reply by djsbriscoe** [reply]
Tried it on the Uno32 board as well and works fine with the same "Blink" example that I always use to test a board. Thanks for your help so far.

David.

**Reply by majenkotech** [reply]
Fantastic. Well, if you're ever wanting more in-depth knowledge of the PIC32 there's plenty of brains here that you can pick any time.

**Reply by djsbriscoe** [reply]
Thanks. Will do. Off to get a well deserved cup of tea. Bye for now.

David.

**Reply by cb88** [reply]
I've been using Chipkit 2.1.0 with Arduino IDE 2.2.1 from the Arduino website not the windows store package. It isn't perfect but is better than the old IDE.
